### PR TITLE
2112: Get average rating from openlist

### PR DESCRIPTION
**Pull request by BirkAndMe**
http://platform.dandigbib.org/issues/2112

### CODE REVIEW

**Comment by BirkAndMe** [comment]
Sure, we'll implement a static cache, setting the rating using a `ding_entity_rating_set_average()` function, instead of using the object as a container.

**Comment by kasperg** [comment]
@BirkAndMe :+1:

File: modules/p2/ding_entity_rating/ding_entity_rating.module [hunk 1]
```diff
@@ -345,14 +345,54 @@ function ding_entity_rating_build_theme_list($rating) {
...
-
-  return $result->fetchField(0);
+  return ding_entity_rating_average($entity);
+}
+
+/**
+ * Set the average rating of a ting object entity.
+ *
+ * @param object $entity
+ *   The ting object entity to remove from the rating list.
```
Comment (kasperg) [inline modules/p2/ding_entity_rating/ding_entity_rating.module]: Nitpicking here but this bit of DocBlock seems copy/pasted. Could you please update it?

File: modules/p2/ding_entity_rating/ding_entity_rating.module [hunk 2]
```diff
@@ -345,14 +345,54 @@ function ding_entity_rating_build_theme_list($rating) {
...
+ */
+function ding_entity_rating_set_average($entity, $value) {
+  return ding_entity_rating_average($entity, $value);
+}
+
+/**
+ * Get / Set average rating of a ting object.
+ *
+ * @param object $entity
+ *   The ting object entity to remove from the rating list.
```
Comment (kasperg) [inline modules/p2/ding_entity_rating/ding_entity_rating.module]: As above.

**Pull request merged**